open_spiel/python/algorithms/nash_averaging.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2019 DeepMind Technologies Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+  Nash averaging.
+  Based on https://arxiv.org/pdf/1806.02643.pdf
+  An axiomatic strategy evaluation metric for 
+  Agent-vs-Agent or Agent-vs-Task two-player zero-sum games
+"""
+
+from cvxopt import solvers, matrix, spdiag, spmatrix, log
+import numpy as np
+
+from open_spiel.python.egt.utils import game_payoffs_array
+
+
+
+def max_entropy_nash(p_mat, eps=0.0):
+  '''
+    Solving for maxent symmetric nash for symmetric two-player zero-sum games
+    convex programming:
+      min p^Tlog(p)
+      s.t. 
+      p_mat.dot(p) <= p^T*p_mat*p
+      p >= 0
+      1^T * p = 1
+
+    Args:
+      p_mat: an N*N anti-symmetric payoff matrix
```

Comment:
Make clear that this is player 1's payoff (ie not a full game payoff).

(I like that you are only including the first player's payoff - it enforces zero-sum and two-player on the API level - it is just nice to make this clear in the docs)